reframe/core/runtime.py

```diff
@@ -204,6 +213,17 @@ def mode(self, name):
         except KeyError:
             raise ConfigError('unknown execution mode: %s' % name) from None
 
+    def next_run(self):
+        self._current_run += 1
+
+    @property
+    def current_run(self):
+        """The current run.
+
+        :type: `integer`
```

Comment:
The documentation here must be expanded a bit. Whatever you put here is exposed to the user API. So what is a run, how it can be useful for a regression test?